app/src/main/kotlin/com/mapbox/vision/teaser/replayer/SessionsAdapter.kt

```diff
@@ -1,42 +1,180 @@
+                    holder.iconChecked.setImageResource(R.drawable.ic_not_checked)
+                    holder.iconChecked.visibility = VISIBLE
+                }
+            }
+        } else {
+            if (item is CameraItem) {
+                holder.itemView.isEnabled = true
+            } else {
+                holder.itemView.background = context.getDrawable(R.drawable.session_info_custom_ripple)
+            }
+            holder.iconChecked.visibility = INVISIBLE
+        }
+    }
+
+    private fun bindViewHolderClickListener(holder: SessionViewHolder, item: AdapterItem) {
+        if (!(item is CameraItem && isMultiSelection)) {
+            holder.itemView.setOnClickListener {
+                if (isMultiSelection) {
+                    handleMultiSelectionClick(holder, item)
+                }
+                if (item is CameraItem) {
+                    clickCameraListener.invoke()
+                } else {
+                    clickSessionListener.invoke(item.name)
+                }
+            }
+        }
+    }
+
+    private fun handleMultiSelectionClick(holder: SessionViewHolder, item: AdapterItem) {
+        if (selectedItems.contains(item.name)) {
+            holder.itemView.setBackgroundColor(backgroundColorTransparent)
+            holder.iconChecked.setImageResource(R.drawable.ic_not_checked)
+            selectedItems.remove(item.name)
+        } else {
+            holder.itemView.setBackgroundColor(backgroundColorSelection)
+            holder.iconChecked.setImageResource(R.drawable.ic_checked)
+            selectedItems.add(item.name)
+        }
+    }
+
+    private fun bindViewHolderLongClick(holder: SessionViewHolder, item: AdapterItem) {
+        if (item !is CameraItem) {
+            holder.itemView.setOnLongClickListener {
+                if (isMultiSelection) {
+                   handleMultiSelectionClick(holder, item)
+                } else {
+                    selectedItems.add(item.name)
+                    onActivateMultiSelectionListener.invoke()
+                }
+                true
+            }
+        }
+    }
 
-        init {
-            textView.setOnClickListener {
-                clickListener.invoke(sessions[adapterPosition])
+    fun activateMultiSelection() {
+        isMultiSelection = true
+        notifyDataSetChanged()
+    }
+
+    fun resetMultiSelection() {
+        isMultiSelection = false
+        selectedItems.clear()
+        notifyDataSetChanged()
+    }
+
+    fun selectAll() {
+        items.forEach {
+            if (it is SessionItem) {
+                selectedItems.add(it.name)
             }
         }
+        notifyDataSetChanged()
+    }
+
+    fun updateSessionsList() {
+        items.clear()
+        items.add(CameraItem(cameraString))
+        File(basePath).listFiles().forEach {
+            val dateString = dateFormatter.format(it.lastModified())
+            items.add(SessionItem(it.name, dateString))
+        }
+    }
```

Comment:
`notifyDataSetChanged()` ?

Reply:
yes!, I've added input param to manage this update